tests/test_bids_file2.m

```diff
@@ -0,0 +1,79 @@
+function test_suite = test_bids_file2 %#ok<*STOUT>
+  try % assignment of 'localfunctions' is necessary in Matlab >= 2016
+    test_functions = localfunctions(); %#ok<*NASGU>
+  catch % no problem; early Matlab versions can use initTestSuite fine
+  end
+  initTestSuite;
+end
+
+function test_validation()
+  assertExceptionThrown(@() bids.File2.validateString([2 3], 'Error', '.*'), '');
+  assertExceptionThrown(@() bids.File2.validateString(['.nii'; '.abc'], 'Error', '.*'), '');
+  assertExceptionThrown(@() bids.File2.validateString('abc', 'Error', 'abcde'), '');
+  bids.File2.validateString('', 'Error', '.*');
+
+  assertExceptionThrown(@() bids.File2.validateExtension('nii'), '');
+  assertExceptionThrown(@() bids.File2.validateExtension('.nii-'), '');
+  assertExceptionThrown(@() bids.File2.validateExtension('.nii_'), '');
+
+  assertExceptionThrown(@() bids.File2.validatePrefix('abc/def'), '');
+  assertExceptionThrown(@() bids.File2.validatePrefix('abcsub-def'), '');
+
+  assertExceptionThrown(@() bids.File2.validateWord('abc/def', 'Word'), '');
+  assertExceptionThrown(@() bids.File2.validateWord('abc-def', 'Word'), '');
+end
+
+function test_parsing()
+
+  filename = 'wuasub-01_ses-test_task-faceRecognition_run-02_bold.nii';
+  file = bids.File2(filename, 'use_schema', false);
+
+  entities = struct('sub', '01', 'ses', 'test', ...
+                    'task', 'faceRecognition', 'run', '02');
+
+  assertEqual(file.prefix, 'wua');
+  assertEqual(file.suffix, 'bold');
+  assertEqual(file.extension, '.nii');
+  assertEqual(file.entities, entities);
+  assertEqual(file.filename, 'wuasub-01_ses-test_task-faceRecognition_run-02_bold.nii');
+  assertEqual(file.json_filename, 'wuasub-01_ses-test_task-faceRecognition_run-02_bold.json');
+  assertEqual(file.bids_path, fullfile('sub-01', 'ses-test'));
+
+  filename = [];
+  filename.prefix = 'wua';
+  filename.suffix = 'bold';
+  filename.ext = '.nii';
+  filename.entities = entities;
+  assertEqual(file.filename, 'wuasub-01_ses-test_task-faceRecognition_run-02_bold.nii');
```

Comment:
I am not clear as to what this is testing. Line 47 is the same as line 38 and the `filename` is not used in this test.

Reply:
Line 47 supposed to tests the creation out of structure instead of filename on line 29. But I forgot to initialize file :
```matlab
file = bids.File2(filename);
``` 

Feel need to write tests for tests files